test/suite/test_cursor_bound_fuzz.py

```diff
@@ -108,129 +118,262 @@ def dump_key_range(self):
+    def apply_truncate(self, session, cursor, cursor2, prepare):
+        lower_key = self.get_random_key()
+        if (lower_key + 1 < self.max_key):
+            upper_key = random.randrange(lower_key + 1, self.max_key)
+            cursor.set_key(lower_key)
+            cursor2.set_key(upper_key)
+            self.assertEqual(session.truncate(None, cursor, cursor2, None), 0)
+
+            for key_id in range(lower_key, upper_key + 1):
+                self.key_range[key_id].update(None, key_states.DELETED, self.current_ts, prepare)
+
+            self.verbose(2, "Truncated keys between: " + str(lower_key) + " and: " + str(upper_key))
+
+    # Each iteration calls this function once to update the state of the keys in the database and
+    # in memory.
+    def apply_ops(self, session, cursor, prepare):
         op_count = self.key_count
-        for i in self.key_range_iter():
-            op = random.choice(list(operations))
-            if (op is operations.UPSERT):
-                self.apply_update(cursor, i)
-            elif (op is operations.REMOVE):
-                self.apply_remove(cursor, i)
+        op = random.choice(list(operations))
+        if (op is operations.TRUNCATE and self.applied_ops):
+            cursor2 = session.open_cursor(self.uri + self.file_name)
+            self.apply_truncate(session, cursor,cursor2, prepare)
+        else:
+            for i in self.key_range_iter():
+                if (random.uniform(0, 1) < self.update_frequency):
+                    continue
+                op = random.choice(list(operations))
+                if (op is operations.TRUNCATE):
+                    pass
+                elif (op is operations.UPSERT):
+                    self.apply_update(cursor, i, prepare)
+                elif (op is operations.REMOVE):
+                    self.apply_remove(cursor, i, prepare)
+                else:
+                    raise Exception("Unhandled operation generated")
+        self.applied_ops = True
+
+    # As prepare throws a prepare conflict exception we wrap the call to anything that could
+    # encounter a prepare conflict in a try except, we then return the error code to the caller.
+    def prepare_call(self, func):
+        try:
+            ret = func()
+        except wiredtiger.WiredTigerError as e:
+            if wiredtiger.wiredtiger_strerror(wiredtiger.WT_PREPARE_CONFLICT) in str(e):
+                ret = wiredtiger.WT_PREPARE_CONFLICT
             else:
-                raise Exception("Unhandled operation generated")
+                raise e
+        return ret
+
+    # Given a bound this key returns what would be the start of the bounded range for that bound.
+    # e.g. if we have a lower bound then the key would be the lower bound, however if the lower
+    # bound isn't enabled then the lowest possible key would be min_key.
+    # max_key isn't inclusive so we subtract 1 off it.
+    def get_expected_key(self, bound_set, type):
```

Comment:
The comment doesn't match the function, since we also deal with returning the end of the key as well. I would also rename the function to fit that is trying to find the expected start and end range of a bound range.